src/components/MultiSelectBase/MultiSelectBase.tsx

```diff
@@ -57,40 +59,25 @@ const MultiselectTextField = React.forwardRef<HTMLInputElement, Props & InputPro
       onClearAllOptions,
       isLoading,
       rightIcon,
+      isInteractive = true,
       ...rest
     } = props;
 
-    const TextfieldRef = React.useRef<HTMLDivElement>(null);
-
     const theme = useTheme();
-
     const hasValue = Boolean(value || (selectedOptions?.length && selectedOptions?.length > 0));
 
-    const inputPlaceholder = useMemo(() => {
-      if (!label && placeholder) {
-        return required ? `${placeholder} *` : placeholder;
-      }
-
-      return label;
-    }, [label, placeholder, required]);
-
-    const iconName = useMemo(() => {
-      if (locked) {
-        return 'lock';
-      }
-
-      if (hasValue) {
-        return 'close';
-      }
-
-      return 'search';
-    }, [hasValue, locked]);
-
-    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
-      if (value === '' && event.key === 'Backspace') {
-        onOptionDelete();
-      }
-    };
+    const { inputPlaceholder, handleKeyDown, icon, hasLabel, TextfieldRef } =
+      useMultiSelectBaseUtils({
+        label,
+        placeholder,
+        required,
+        locked,
```

Comment:
for some reason based on our rules I expect this to be `isLocked`

Also I suggest hooks are made for multi filter use? Because here is used only once, therefore makes it obsolete